### ISSUE TITLE
About the results on PASCAL Context

### ISSUE THREAD

**Issue by qiulesun** [junfu1115/DANet#11]
**Question 1:** 
The DANet performs well on PASCAL Context dataset, outperforming EncNet by 1%. Do that use any improvement strategies mentioned in the subsection 'Ablation Study for Improvement Strategies' ?  
As far as I known, the EncNet achieves 51.7 mIoU on the same dataset without using multiple scales strategy.
**Question 2:** 
Before conducting softmax operation on x, we usually subtract the maximum value of x. But for CAM_Module, I see that energy_new = torch.max(energy, -1, keepdim=True)[0].expand_as(energy)-energy (https://github.com/junfu1115/DANet/blob/master/encoding/nn/attention.py#L75) rather than energy_new = energy- torch.max(energy, -1, keepdim=True)[0].expand_as(energy) followed by softmax op. Therefore, the weights of each channel are reverse based on the two different op before softmax.
Can you share why the former is used?

**Comment by junfu1115** [comment]
Q#1, we report the results on 60 classes, while EncNet achieves 51.7 mIoU on 59 classes w/o background.
Q#2, we didn't try the latter one.

**Comment by junfu1115** [comment]
Q#1, Yes
Q#2, We don't try sigmoid, in fact, we think softmax is more appropriate since it involves all pixels

**Comment by TensorDL** [comment]
**Question 1:**
What's the major difference between PAM with non-local (https://arxiv.org/pdf/1711.07971.pdf)?

**Comment by qiulesun** [comment]
**Question 1:**
I am sorry for bothering you again. What is the base size and crop size of PContext and COCO Stuff datasets for traing and multiscale test?

**Comment by junfu1115** [comment]
On the PContext dataset, the base size is set to 608 and crop size is set to 576. Multi-scale test is adopted, you can refer to the experimental settings of  [PyTorch-Encoding](https://github.com/zhanghang1989/PyTorch-Encoding)

**Comment by HolmesShuan** [comment]
@qiulesun @junfu1115 Hi, I have mailed the authors of CCL(i.e., [6] in DANet). The result of [6] in Table 6 is actually 59 classes without background and with single scale testing. It may be unfair to compare with CCL directly :(

**Comment by qiulesun** [comment]
@junfu1115 [trivial questions and suggestions #179](https://github.com/zhanghang1989/PyTorch-Encoding/issues/179) and [Performance on Pascal Context #78](https://github.com/HRNet/HRNet-Semantic-Segmentation/issues/78) give different strategies to compute mIoU on 60 classes for PASCAL Context.  They confused me. How does DANet calculate the mIoU on 60 classes ?

**Comment by qiulesun** [comment]
Q1: In DANet, could you provide the settings of base size and crop size for each dataset ?
Q2: CAM_Module computes the [energy_new](https://github.com/junfu1115/DANet/blob/master/encoding/nn/da_att.py#L75) before softmax while PAM_Module not, why ?
